### ISSUE TITLE
Unable to run image with > 128 layers. Max depth exceeded

### ISSUE THREAD

**Issue by rstanevich** [cri-o/cri-o#6261]
### What happened?

The error happens when deploying a pod with image having > 128 layers 
```
message: 'failed to mount container xxxxxxx: max depth exceeded'
reason: CreateContainerError
```

### What did you expect to happen?

The Pod should be launched successfully

### How can we reproduce it (as minimally and precisely as possible)?

1. `minikube start --driver qemu2 --container-runtime=crio`
2. Build and deploy to k8s an image having > 128 layers

### Anything else we need to know?

`containerd` doesn't have such limitation

### CRI-O and Kubernetes version

<details>

```console
$ crio --version
1.24.0
```

```console
$ kubectl --version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-16T15:58:47Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:38:15Z", GoVersion:"go1.19", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME=Buildroot
VERSION=2021.02.12-1-g1be7c81-dirty
ID=buildroot
VERSION_ID=2021.02.12
PRETTY_NAME="Buildroot 2021.02.12"
$ uname -a
Linux minikube 5.10.57 #1 SMP Sat Sep 10 02:24:46 UTC 2022 x86_64 GNU/Linux
```

</details>

### Additional environment details (AWS, VirtualBox, physical, etc.)

<details>

</details>

**Comment by haircommander** [comment]
hm I wonder if we can update it. It seem to be ensuring the link directory doesn't fail because of length: https://github.com/containers/storage/blob/e40d88e9d49b9596547f6e7a292dfa76ea3f0ddd/drivers/overlay/overlay.go#L67-L78

@nalind @rhatdan @giuseppe is there anything we can do here?

**Comment by rstanevich** [comment]
Thank you, @haircommander!

I would like to share the original context - https://github.com/moby/buildkit/issues/3067
We can build an OCI image using `buidah`, `buildkit` or `podman` with any number of layers.
But at runtime, the overlay (I think it's the recommended storage driver) has a hard limit of 128 layers.

Therefore, I'm looking for an idiomatic way to control the situation so that I don't get into an unexpected failure when deploying an image. And of course, I can control the number of layers at the build step.

**Comment by nalind** [comment]
> hm I wonder if we can update it. It seem to be ensuring the link directory doesn't fail because of length: https://github.com/containers/storage/blob/e40d88e9d49b9596547f6e7a292dfa76ea3f0ddd/drivers/overlay/overlay.go#L67-L78
> 
> @nalind @rhatdan @giuseppe is there anything we can do here?

The kernel won't let us try to use more than 500 lower directories.  That's a hard limit, and we might need to add one or two on top of the ones that are part of an image in order to create a container's rootfs, but we could try to handle that many.

**Comment by haircommander** [comment]
this will be fixed in 1.26.0 and 1.25.2
